strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/providers/io/MavenMetadataExpiredRepositoryPathHandler.java

```diff
@@ -0,0 +1,112 @@
+package org.carlspring.strongbox.providers.io;
+
+import org.carlspring.strongbox.providers.repository.ProxyRepositoryProvider;
+import org.carlspring.strongbox.providers.repository.RepositoryProvider;
+import org.carlspring.strongbox.providers.repository.RepositoryProviderRegistry;
+import org.carlspring.strongbox.storage.checksum.ChecksumCacheManager;
+import org.carlspring.strongbox.storage.metadata.MetadataHelper;
+import org.carlspring.strongbox.storage.repository.Repository;
+
+import javax.inject.Inject;
+import java.io.IOException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import static org.carlspring.strongbox.providers.io.MavenMetadataExpiredRepositoryPathHandler.Decision.I_DONT_KNOW;
+import static org.carlspring.strongbox.providers.io.MavenMetadataExpiredRepositoryPathHandler.Decision.NO_LEAVE_IT;
+import static org.carlspring.strongbox.providers.io.MavenMetadataExpiredRepositoryPathHandler.Decision.YES_FETCH;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@Component
+public class MavenMetadataExpiredRepositoryPathHandler
+        implements MavenExpiredRepositoryPathHandler
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(MavenMetadataExpiredRepositoryPathHandler.class);
+
+    @Inject
+    private RepositoryProviderRegistry repositoryProviderRegistry;
+
+    @Inject
+    private RepositoryPathResolver repositoryPathResolver;
+
+    @Inject
+    private ChecksumCacheManager checksumCacheManager;
```

Comment:
Sorry, but like I have told earlier we shouldn't use `ChecksumCacheManager` within this issue because it have another purpose. It was assumed to be used within artifact uploading to check that artifact matched checksum. 
As about your observation about 'DEBUG' message, then there was a task to make configurable strict checksum validation for uploaded artifacts.
Also this `ChecksumCacheManager` was developed by @carlspring, so maybe he can say something more about it.

Reply:
I think this "new coat" of `ChecksumCacheManager` may be useful for both purposes. Besides that, new implementation follows the approach to work on `RepositoryPath`  (and `URI`) instead of `String`s. This PR bases on `ChecksumCacheManager`.

Reply:
> Besides that, new implementation follows the approach to work on RepositoryPath (and URI) instead of Strings
this is very good, and we should keep this `RepositoryPath` improvement, but can you please clarify why you need `ChecksumCacheManager` within `maven-metadata.xml` invalidation?

Reply:
To cache `maven-metadata.xml` checksum

Reply:
why you need to cache it?

Reply:
Some tools may not send `.sha1` or `.md5` next to the `maven-metadata.xml`. New `ChecksumCacheManager` stores checksums calculated in runtime (using our streams) as well as `.sha1` or `.md5` values if provided.

Without it algorithm described in #827 will be much more inefficient if we no't have checksums (we will be always re-fetching `maven-metadata.xml`).

Reply:
This solution would be fine for `maven-metadata.xml` and maven in general. I think we should consider more generic approach (like cache or database). I choosed the cache for the time being.

Reply:
I believe that if the `maven-metadata.xml` file has not yet been aggregated in a group, attempt to download it from all of the repositories in the group and create a merged representation. This should also be done each time the metadata expires.

Reply:
> I believe that if the maven-metadata.xml file has not yet been aggregated in a group, attempt to download it from all of the repositories in the group and create a merged representation. This should also be done each time the metadata expires.

also, this will probably better to do in parallel so we have no performance issues

Reply:
> This solution would be fine for maven-metadata.xml and maven in general. I think we should consider more generic approach (like cache or database). I choosed the cache for the time being.

why do you think so? as for me it will be generic enough.

Anyway I am not sure that your approach will work properly.

If you want to do it this way then please think how you can get checksum from `InputStream` without using `ChecksumCacheManager`.

@carlspring we probably should rename `ChecksumCacheManager` to make it clear that it is relevant only for upload checksum validation.

Reply:
@carlspring do you have some proposal to rename currently existing [ChecksumCacheManager ](https://github.com/strongbox/strongbox/blob/master/strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/storage/checksum/ChecksumCacheManager.java)? I'll probably roll it back. Also, I'd like to gain from my implementation too as it's a real cache (please let me know if you disagree).

Reply:
I hadn't thought of this. I think the name is okay, despite the fact that it's only used during uploads... @fuss86 , if we can benefit from your changes, we should!